app/Models/Question.php

```diff
@@ -220,4 +223,35 @@ public function hashtags(): BelongsToMany
     {
         return $this->belongsToMany(Hashtag::class);
     }
+
+    /**
+     * Run the parser for the given attribute and value.
+     */
+    private function runParser(?string $value, string $attribute): ?string
+    {
+        $key = "question.{$this->id}.{$attribute}.parsed";
+
+        if (Context::hasHidden($key)) {
+            if ($this->isDirty($attribute)) {
+                return $this->cacheAndParse($value, $key);
+            }
+
+            $returned = Context::getHidden($key);
+
+            return is_string($returned) ? $returned : null;
+        }
+
+        return $this->cacheAndParse($value, $key);
+    }
+
+    /**
+     * Cache and parse the given value.
+     */
+    private function cacheAndParse(?string $value, string $key): ?string
+    {
+        $parsed = (new ParsableContent)->parse((string)$value);
```

Comment:
Can you hold a "static" array within `ParsableContent` class itself? Be sure to flush the array within an service provider after...

Saying this because, context is used for information, not for caching.

Reply:
cc @CamKem

Reply:
Hey @nunomaduro, sorry I've been busy with work.

I figured as the Context is a request persistent object for `information` state management, it would be ideal to utilise for in-memory caching within the current request. I think it's functionally effectively the same as using a static array, but I understand your point.

I was initially going to use a static property on the `ParsableContent` class, but as it involved additional changes including ignoring the class in the mutation the arch tests, due to removing the readonly modifier on the class, I went with Context.

Anyhow, I have made the requested changes, including a service provider for the termination flush & added tests for coverage. Let me know if its all ok.